hs/src/Reach/JSUtil.hs

```diff
@@ -267,19 +317,63 @@ instance HasJSAnnot JSExpression where
     JSVarInitExpression a _ -> jsa a
     JSYieldExpression a _ -> a
     JSYieldFromExpression a _ _ -> a
+  rjsa b a' = case b of
+    JSIdentifier _a x -> JSIdentifier a' x
+    JSDecimal _a x -> JSDecimal a' x
+    JSLiteral _a x -> JSLiteral a' x
+    JSHexInteger _a x -> JSHexInteger a' x
+    JSOctal _a x -> JSOctal a' x
+    JSStringLiteral _a x -> JSStringLiteral a' x
+    JSRegEx _a x -> JSRegEx a' x
+    JSArrayLiteral _a x y -> JSArrayLiteral a' x y
```

Comment:
You're going to have to recur inside everything; also `y` is ALSO an annotation (it's the right `]`)